### PR TITLE
feat(api-client): add default queries override

### PR DESCRIPTION
**Pull request by marc-arnoult**
Add the possibility to override default queries.

## Description

Currently to be able to override one single field in the entire application we have to use https://docs.vuestorefront.io/magento/guide/composables/use-custom-query.html 

The issue are :
- Sending a huge payload ( graphql ).
- Have to potentially copy and override the logic of the initial composable.
- Override everywhere the default query is used.

To remove a simple field on the GraphQL query, this is not optimal and can be even harder to maintain the code for the future version upgrade in the theme section.

## Related Issue
https://github.com/vuestorefront/magento2/issues/245

## Motivation and Context

Currently we do not use multiple fields everywhere, ( `reviews` in products query, `allow_remote_assitance` in customer query.... )
To be able to override it, we have to copy/paste all the logic from login/registration composable for example because there are some queries called inside these composables.

## How Has This Been Tested?
- Every API endpoint called with default and overrided query.
- Home page
- Category page
- Product page
- Login
- Registration
- Wishlist ( add/remove )
- Cart ( add/update/remove )
- Login
- Registration
- My account page / Sub section
- Logout

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by marc-arnoult** [comment]
@Frodigo Hi, i would like to add some tests on every endpoints.
Tell me if you want it in this PR or later once https://github.com/vuestorefront/magento2/issues/279 merged maybe.

**Comment by bloodf** [comment]
@marc-arnoult for the tests, I think it's good to have another PR, so we can have more controllable PRs each time.

Congratulations on the implementation :)

**Review by bloodf — APPROVED** [review]

File: docs/guide/override-queries.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Override queries
```
Comment (bloodf) [inline docs/guide/override-queries.md:1]: Have you checked if the docs is getting this page?

Comment (marc-arnoult) [reply docs/guide/override-queries.md:1]: @bloodf  Good catch ! Fixed.

**Comment by marc-arnoult** [comment]
> @marc-arnoult for the tests, I think it's good to have another PR, so we can have more controllable PRs each time.
> 
> Congratulations on the implementation :)

Very well, thank you !

**Review by Frodigo — APPROVED** [review]
Hi, @marc-arnoult looks awesome. Thanks for your contribution! 

Regarding tests, I let you know when it will be possible. 

Thanks,
Marcin

**Pull request merged**